### PR TITLE
[KYUUBI #1123][TEST] Split KyuubiExtensionSuite according to different extensions

### PR DESCRIPTION
**Pull request by simon824**
split KyuubiExtensionSuite according to different extensions
#1123

drop：
`KyuubiSparkExtensionTest`

add：
- `FinalStageConfigIsolationSuite`
- `InsertShuffleNodeBeforeJoinSuite`
- `WatchDogSuite`
- `RepartitionBeforeWriteSuite`
- `SqlClassificationSuite`

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
can we rename the suffix Test to Suite?

**Comment by simon824** [comment]
> can we rename the suffix Test to Suite?

Done

File: dev/kyuubi-extension-spark-3-1/src/test/scala/org/apache/spark/sql/FinalStageConfigIsolationSuite.scala [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.sql.execution.adaptive.{CustomShuffleReaderExec, QueryStageExec}
+import org.apache.spark.sql.internal.SQLConf
+
+import org.apache.kyuubi.sql.{FinalStageConfigIsolation, KyuubiSQLConf}
+
+class FinalStageConfigIsolationTest extends KyuubiSparkSQLExtensionTest {
```
Comment (ulysses-you) [inline dev/kyuubi-extension-spark-3-1/src/test/scala/org/apache/spark/sql/FinalStageConfigIsolationSuite.scala:25]: don't forget rename class name

Comment (simon824) [reply dev/kyuubi-extension-spark-3-1/src/test/scala/org/apache/spark/sql/FinalStageConfigIsolationSuite.scala:25]: Done

Comment (ulysses-you) [reply dev/kyuubi-extension-spark-3-1/src/test/scala/org/apache/spark/sql/FinalStageConfigIsolationSuite.scala:25]: have you updated `FinalStageConfigIsolationTest` ?

Comment (cfmcgrady) [reply dev/kyuubi-extension-spark-3-1/src/test/scala/org/apache/spark/sql/FinalStageConfigIsolationSuite.scala:25]: It looks like you missed this one `FinalStageConfigIsolationTest`.

**Review by cfmcgrady — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**